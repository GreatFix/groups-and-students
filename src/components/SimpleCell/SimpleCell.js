import React from 'react'
import classes from './SimpleCell.module.css'
import PropTypes from 'prop-types'

const SimpleCell = (props) => {
  return (
    <p
      className={`${classes.SimpleCell} ${props.className} ${props.onClick ? classes.Hover : ''}`}
      onClick={props.onClick}
    >
      {props.children}
    </p>
  )
}

SimpleCell.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func,
}

SimpleCell.defaultProps = {
  className: '',
  children: null,
  onClick: null,
}

export default React.memo(SimpleCell)
