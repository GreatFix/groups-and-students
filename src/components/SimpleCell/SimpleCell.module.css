.SimpleCell {
  display: grid;
  justify-content: center;
  align-content: center;
  background-color: #41f0b0;
  border-radius: 2px;
  font-size: 1em;
  font-family: 'Roboto Condensed', sans-serif;
  padding: 5px;
  margin: 4px 2px 0 2px;
}
.Hover:hover {
  background-color: #41f041;
}
