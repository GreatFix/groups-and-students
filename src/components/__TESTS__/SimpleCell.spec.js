import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import userEvent from '@testing-library/user-event'
import SimpleCell from '../SimpleCell/SimpleCell'

const TEST_TEXT = 'May the force be with you'
const TEST_CLASS = 'TEST_CLASS'

describe('SimpleCell', () => {
  test('renders a text', () => {
    const { getByText } = render(<SimpleCell>{TEST_TEXT}</SimpleCell>)
    expect(getByText(TEST_TEXT)).toBeInTheDocument()
  })

  test('gets a class', () => {
    const { container } = render(<SimpleCell className={TEST_CLASS}>{TEST_TEXT}</SimpleCell>)
    expect(container.querySelector('.SimpleCell')).toHaveClass(TEST_CLASS)
  })

  test('onClick with onClick', () => {
    const handleClick = jest.fn()
    const { container } = render(<SimpleCell onClick={handleClick}>{TEST_TEXT}</SimpleCell>)
    userEvent.click(container.querySelector('.SimpleCell'))
    expect(handleClick.mock.calls.length).toBe(1)
  })

  test('render properly', () => {
    const { container } = render(<SimpleCell>{TEST_TEXT}</SimpleCell>)
    expect(container).toMatchSnapshot()
  })
})
