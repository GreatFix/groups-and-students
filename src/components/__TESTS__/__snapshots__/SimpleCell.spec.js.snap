// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleCell render properly 1`] = `
<div>
  <p
    class="SimpleCell  "
  >
    May the force be with you
  </p>
</div>
`;
